### DISCUSSION TITLE
Keyboard hook

### DISCUSSION THREAD

**Discussion by d-bug** [kimmknight/remoteapptool#127] [Q&A]
keyboard language change (Shift+Alt) does not work

Client: Windows 10 Pro and Windows 11 Home
Server: Windows Server 2022 Datacenter (Azure based VM)

I have tried all methods 0,1,2 with no success?

Any suggestions will be welcome

Thank you

**Comment by MrBrianGale** [comment]
As a guess - do you have that hotkey configured on the SERVER?
There is no way to add what you requested to the remoteapptool as that's not what's running when you are doing stuff with the remote app. RemoteAppTool is just a frontend and configuration tool for RemoteApps which is a part of the RDP protocol. The problem MAY be that the server isn't configured to capture those hotkeys OR it could be your RDP application doesn't properly capture "shift+alt".
ALL remote apps run on the SERVER, so anything configured and set up locally is not being captured. If, for example, you set the date on the server to be 2025 instead of 2024, all applications run from the remote app will see the date as 2025 even though your local machine is set to 2024.

**Reply by d-bug** [reply]
Hello and thank you for your reply.

As a note I would like to mention that when I run the RDP client application (Microsoft) to connect from workstation to VM desktop and run the app from there, the key combinations works properly.
When th app is opened through RemoteApp tool the key combination does not work. (is not captured in the app - not in VM)

Thank you.

**Reply by kimmknight** [reply]
You could try setting the "keyboardhook" option to 1 in RDP Options when exporting the RDP file.

It probably won't cause shift-alt to work for you, but it might allow the alternate switch hotkey to work: WINDOWS-SPACE

**Comment by kimmknight** [comment]
I tested this, going from a Win 10 client to a Win 10 host and Windows keyboard layout switching just works for me. When I'm using the RemoteApp and press SHIFT-ALT the language definitely switches for the remote session.

This can be confirmed this by enabling the Language Bar in "Floating on Desktop" mode on the remote host PC. (Be sure to disable this after testing as the language bar seems to stay open after all RemoteApps are closed, preventing the session from terminating).

I also note that when using the RemoteApp, the language/keyboard-switch icon/button in my (local) system tray (next to the clock) gets hidden. When I switch to a locally installed app the icon reappears.
